src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArray = [];
+
+  let stateCopy = {
+    ...state,
+  };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        stateArray.push({ ...stateCopy });
```

Comment:
looks like `stateArray.push({ ...stateCopy });` repeat each time, so it's better to move it down, after `switch`